### DISCUSSION TITLE
How i open the book in position saved in preference

### DISCUSSION THREAD

**Discussion by malainho** [readium/kotlin-toolkit#472] [Q&A]
Hello People,

I use this lib to view a book in format .epub in my app:

implementation("org.readium.kotlin-toolkit:readium-shared:2.4.0")
implementation("org.readium.kotlin-toolkit:readium-streamer:2.4.0")
implementation("org.readium.kotlin-toolkit:readium-navigator:2.4.0")
implementation("org.readium.kotlin-toolkit:readium-opds:2.4.0")
implementation("org.readium.kotlin-toolkit:readium-lcp:2.4.0")

I am save a current position in Preference.

It's ok. I verify this. Now close the app, for example.
When i open the book i want open the book in position saved in preference.
I try this, but open the book in initial position:

What i do to open the book in position saved before ?

Thanks

![1707239343949](https://github.com/readium/kotlin-toolkit/assets/3917781/17f0a4a0-5940-447b-969d-4ac0467d221b)
![1707239560713](https://github.com/readium/kotlin-toolkit/assets/3917781/d35c0814-a598-44d3-b1d0-35d0d881f32e)

**Comment by mickael-menu** [comment]
Take a look at this user guide:
https://github.com/readium/kotlin-toolkit/blob/documentation/docs/guides/navigator/navigator.md#saving-and-restoring-the-reading-progression

**Reply by malainho** [reply]
Hello Mickael,

Tank you very mutch for the link but i don't know i apply the canges.
Can i send for you my code of my Fragment ?
You can help me ?
That is the only thing that is missing to finish my Application.

Thank you very mutch

**Reply by mickael-menu** [reply]
I'm sorry, we don't offer that kind of support.

**Comment by malainho** [comment]
Hello, my code for save the position of book is correct ?
